framework/tlvf/src/include/tlvf/BaseClass.h

```diff
@@ -38,6 +37,19 @@ class BaseClass {
     const bool m_parse__;
     const bool m_swap__;
     bool m_init_succeeded = false;
+
+private:
+    /**
+     * @brief trim buffer by tailroom size
+     *
+     * This is intended to be used __only__ by friend class TlvList on finalize
+     * When resizing buffers.
+     *
+     * @param tailroom size to trim
+     * @return true on success
+     * @return false on failure
+     */
+    friend void trim(BaseClass &obj, size_t tailroom) { obj.m_buff_ptr__ -= tailroom; };
```

Comment:
A friend function of a class is defined outside of a class scope, so I think we should use the friend keyword in the TlvList class, and declare the trim method as a friend method of TlvList class, so it has the right to access all private and protected members of the class and use this method.

Reply:
Nevermind, that is not going to work since `trim` is a private class method, but still making `trim` a friend method is wrong since it can be used by other classes, for example, CmduMessageTx, and that was not Tomer's intention.